api/institutions/authentication.py

```diff
@@ -27,42 +55,45 @@ def authenticate(self, request):
         except (jwt.InvalidTokenError, TypeError):
             raise AuthenticationFailed
 
-        # The JWT `data` payload is expected in the following structure.
-        #
-        # {"provider": {
-        #     "idp": "https://login.circle.edu/idp/shibboleth",
-        #     "id": "CIR",
-        #     "user": {
-        #         "middleNames": "",
-        #         "familyName": "",
-        #         "givenName": "",
-        #         "fullname": "Circle User",
-        #         "suffix": "",
-        #         "username": "user@circle.edu"
-        #     }
-        # }}
         data = json.loads(payload['data'])
         provider = data['provider']
 
         institution = Institution.load(provider['id'])
         if not institution:
             raise AuthenticationFailed('Invalid institution id specified "{}"'.format(provider['id']))
 
-        username = provider['user']['username']
-        fullname = provider['user']['fullname']
+        username = provider['user'].get('username')
+        fullname = provider['user'].get('fullname')
+        given_name = provider['user'].get('givenName')
+        family_name = provider['user'].get('familyName')
+
+        # use given name and family name to build full name if not provided
+        if given_name and family_name and not fullname:
```

Comment:
OSF requires `fullname` when creating a user. If `fullname` is not provided:
1. construct `fullname` from `given_name` and `family_name`